cdtb/binfile.py

```diff
@@ -211,124 +211,110 @@ class BinType(IntEnum):
     HASH = 17
     PATH = 18  # introduced in 10.23
     # Complex types (0x80 flag introduced in 9.23)
-    CONTAINER = 0x80
-    CONTAINER2 = 0x81  # introduced in 10.8
+    LIST = 0x80
+    LIST2 = 0x81  # introduced in 10.8
     STRUCT = 0x82
     EMBEDDED = 0x83
     LINK = 0x84
     OPTION = 0x85
     MAP = 0x86
     FLAG = 0x87
 
+def format_binvalue(btype, bvalue):
+    if btype in (BinType.LIST, BinType.LIST2):
+        svalues = _repr_indent_list(bvalue.values)
+        return f"{btype.name}({bvalue.vtype.name}) {svalues}"
+    elif btype == BinType.STRUCT:
+        if bvalue.type.h == 0:
+            return f"STRUCT {None}"
+        sfields = _repr_indent_list(bvalue.fields)
+        return f"STRUCT {bvalue.type!r} {sfields}"
+    elif btype == BinType.EMBEDDED:
+        sfields = _repr_indent_list(bvalue.fields)
+        return f"EMBEDDED {bvalue.type!r} {sfields}"
+    elif btype == BinType.OPTION:
+        svalue = None if bvalue.value is None else f'{bvalue.value!r}'
+        return f"OPTION({bvalue.vtype.name}) {svalue}"
+    elif btype == BinType.MAP:
+        svalues = ''.join(f"  {k} => {_repr_indent(v)}\n" for k, v in bvalue.values.items())
+        return f"MAP({bvalue.ktype.name},{bvalue.vtype.name}) {{\n{svalues}}}"
+    else:
+        return f"{btype.name} {bvalue!r}"
 
-class BinStruct(BinObjectWithFieldsAndType):
-    """Structured binary value"""
-
-    def __repr__(self):
-        sfields = _repr_indent_list(self.fields)
-        return f"<STRUCT {self.type!r} {sfields}>"
-
-class BinEmbedded(BinObjectWithFieldsAndType):
-    """Embedded binary value"""
-
-    def __repr__(self):
-        sfields = _repr_indent_list(self.fields)
-        return f"<EMBEDDED {self.type!r} {sfields}>"
-
-class BinField:
-    """Base class for binary fields
-
-    A field is a value (possibly nested) associated to a hash.
-    """
-    def __init__(self, hname):
-        self.name = BinFieldName(hname)
-
-class BinBasicField(BinField):
-    """Binary field for fixed-width, non-nested values"""
 
-    def __init__(self, hname, btype, value):
-        super().__init__(hname)
-        self.type = btype
-        self.value = value
+class BinList():
```

Comment:
```suggestion
class BinList:
```
Same for other classes